File: src/main/java/io/klbz/curie/Unit.java
package io.klbz.curie;

public final class Unit {
	public static final Unit unit = new Unit();

	private Unit(){}

	public static Unit unit(){ return unit; }

	@Override
	public boolean equals(Object obj){ return this == obj; }

	@Override
	public int hashCode(){ return 0; }

	@Override
	public String toString(){ return "Unit"; }
}

File: src/main/java/io/klbz/curie/Unit.kt
package io.klbz.curie

class Unit private constructor() {
    override fun equals(other: Any?) = this === other

    override fun hashCode() = 0

    override fun toString() = "Unit"

    companion object {
        @JvmStatic
        val unit = Unit()

        @JvmStatic
        fun unit() = unit
    }
}
